index.js

```diff
@@ -607,6 +608,15 @@ class ClientServiceBase {
       );
     }
   }
+
+  /**
+   * @param {String|Object} key
+   * @return {Object}
+   */
+  _createSigner(key) {
+    this.signer = this.signer || this.signerFactory.create(key);
+    return this.signer;
+  }
```

Comment:
Sorry, I did not realize the properties needed to be passed to the signer when I made the suggestion earlier. Could this be refactored even further like the following ?
```suggestion
  _createSigner(properties) {
    const key = properties.getPrivateKeyCryptoKey() || properties.getPrivateKeyPem();
    this.signer = this.signer || this.signerFactory.create(key);
    return this.signer;
  }
```

Reply:
No problem.

Reply:
Revised !